static/app/components/searchSyntax/parser.tsx

```diff
@@ -878,6 +890,82 @@ export class TokenConverter {
   };
 }
 
+function parseDate(input: string): Date {
+  const date = new Date(input);
+
+  if (isNaN(date.getTime())) {
+    throw new Error('Invalid date');
+  }
+
+  return date;
+}
+
+function parseRelativeDate(
+  input: string,
+  {sign, unit}: {sign: '-' | '+'; unit: string}
+): Date {
+  let date = new Date(input).getTime();
+
+  if (isNaN(date)) {
+    throw new Error('Invalid date');
+  }
+
+  let offset: number | undefined;
+  switch (unit) {
+    case 'm':
+      offset = 1000 * 60;
+      break;
+    case 'h':
+      offset = 1000 * 60 * 60;
+      break;
+    case 'd':
+      offset = 1000 * 60 * 60 * 24;
+      break;
+    case 'w':
+      offset = 1000 * 60 * 60 * 24 * 7;
+      break;
+    default:
+      throw new Error('Invalid unit');
+  }
+
+  if (offset === undefined) {
+    throw new Error('Unreachable');
+  }
+
+  date = sign === '-' ? date - offset : date + offset;
+  return new Date(date);
```

Comment:
Should this actually produce a moment `Difference` object?

Reply:
I dont think so, I'd rather just have it produce a point in time, the usage is likely going to be a comparison of getTime() between two dates. I wouldnt want to force the usage of moment here

Reply:
We could also maybe just represent this as + - milliseconds. Then users if they want it can do

```tsx
new Date(new Date().getTime() + difference)
```

Reply:
Since the token has a date type, I'd rather represent it as a date tbh